lib/xml/rexml_ext.rb

```diff
@@ -152,7 +152,12 @@ def each_on_axis(axis, &block)
   # temporarily suppress warnings
   class <<Kernel
     alias_method :old_warn, :warn
-    def warn(*args)
+    if RUBY_VERSION >= '2.5.0' # REXML uses additional hash parameters in ruby 2.5 and onwards
```

Comment:
Do we need the if-else here? `def warn(msg, **options)` should be fully compatible with pre-2.5 clients that don't pass any options, afaics.